src/python/gudhi/random_point_generators.cc

```diff
@@ -0,0 +1,68 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Hind Montassif
+ *
+ *    Copyright (C) 2021 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+
+#include <gudhi/random_point_generators.h>
+
+#include <CGAL/Epick_d.h>
+
+namespace py = pybind11;
+
+
+typedef CGAL::Epick_d< CGAL::Dynamic_dimension_tag > Kern;
+
+template <typename Kernel>
+py::array_t<double> generate_points_on_sphere(py::object num_points, py::object dim, py::object radius) {
+    int npoints = num_points.cast<int>();
```

Comment:
Ok, but we still need to cast num_points to int when specifying the _points_ array dimensions, and dim to size_t in the second   loop of copy (unless j is declared as int)

Reply:
I see what you mean. For i and j, I guess we can use the same type as their respective upper bounds.
The shape of an array uses type py::ssize_t (a signed version of size_t). Maybe we could take num_points in that type, and then we wouldn't need to cast it?

Reply:
Ah, no they seem to have used a template initializer_list in pybind11 (why though?), so we need to use exactly the same type for the 2 arguments, so probably we still need to cast dim to py::ssize_t, or maybe simpler also take the dim argument as py::ssize_t instead of int...

Reply:
Since the original generate function uses num_points as size_t and dim as int, I would rather leave it as it is and use the same type for each one in the loops as you suggested. When specifying the py::array dimensions though, maybe cast dim to size_t ?

Reply:
ok.